### PR TITLE
Updated bounty ui

### PR DESCRIPTION
**Pull request by Devorein**
[ x ] Create a bounty
[ x ] Be able to edit bounty from the new UI
[ x ] "View mode" display for bounties from the card view
[ x ] Reviewer Mode: view list of applications
[ x ] Create form to apply to a bounty and update submission

### CODE REVIEW

File: components/bounties/[bountyId]/BountyDetails.tsx [hunk 1]
```diff
@@ -11,14 +11,29 @@ import { usePageTitle } from 'hooks/usePageTitle';
...
   const [, setPageTitle] = usePageTitle();
-  const { currentBounty, currentBountyId } = useBounties();
+  const { currentBounty, setCurrentBounty, currentBountyId, updateCurrentBountyId } = useBounties();
 
   const [bountyPermissions, setBountyPermissions] = useState<AssignedBountyPermissions | null>(null);
 
+  useEffect(() => {
+    if (bounty) {
+      updateCurrentBountyId(bounty.id);
+      setCurrentBounty(bounty);
```
Comment (mattcasey) [inline components/bounties/[bountyId]/BountyDetails.tsx:33]: this logic already exists in [bountyId].tsx, and makes more sense to be higher up

File: components/bounties/[bountyId]/components/BountyApplicantList.tsx [hunk 2]
```diff
@@ -16,27 +15,27 @@ import { Modal } from 'components/common/Modal';
...
   permissions: AssignedBountyPermissions
+  showHeader?: boolean
 }
 
 export function BountyApplicantList ({
   applications,
   bounty,
-  permissions
+  permissions,
+  showHeader = true
```
Comment (mattcasey) [inline components/bounties/[bountyId]/components/BountyApplicantList.tsx:38]: this component is only used in one place, there's no need for a showHeader prop

File: components/bounties/[bountyId]/components_v3/BountyApplicationForm.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import { Box, Button, Stack, Tooltip } from '@mui/material';
```
Comment (mattcasey) [inline components/bounties/[bountyId]/components_v3/BountyApplicationForm.tsx:1]: please move everything to components/ - any reason you left the _v3 in there?

File: components/common/Page/CreatePageButton.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  const [currentSpace] = useCurrentSpace();
+  const [page, setPage] = useState<Page | null>(null);
+  const [currentUserPermissions] = useCurrentSpacePermissions();
+  const suggestBounties = currentUserPermissions?.createBounty === false;
+  const { setBounties } = useBounties();
+
+  async function onClickCreate () {
+    if (currentSpace && user) {
+      if (type === 'proposal') {
+        const newPage = await addPage({
```
Comment (mattcasey) [inline components/common/Page/CreatePageButton.tsx:29]: i don't think it was worth making proposal part of this button. there's so many conditional checks and there's a bunch of stuff related to bounties in here that wasn't necessary for proposals

File: prisma/schema.prisma [hunk 5]
```diff
@@ -389,13 +389,9 @@ model Bounty {
   spaceId           String             @db.Uuid
   createdAt         DateTime           @default(now())
   updatedAt         DateTime           @default(now())
-  description       String
-  descriptionNodes  Json
   rewardAmount      Float
   rewardToken       String
   status            BountyStatus       @default(open)
-  title             String
-  linkedTaskId      String?            @db.Uuid
```
Comment (mattcasey) [inline prisma/schema.prisma:398]: i would have done this in a separate PR tbh

File: components/bounties/components/BountyCard.tsx [hunk 6]
```diff
@@ -34,10 +36,10 @@ function BountyCardDetails ({ bounty, truncate }: Pick<IBountyInput, 'bounty' |
           justifyContent: 'space-between'
         }}
       >
-        <CardHeader title={bounty.page?.title} titleTypographyProps={{ sx: { fontSize: '1rem', fontWeight: 'bold' } }} />
+        <CardHeader title={bountyPage?.title} titleTypographyProps={{ sx: { fontSize: '1rem', fontWeight: 'bold' } }} />
```
Comment (mattcasey) [inline components/bounties/components/BountyCard.tsx:39]: should be "Untitled" if no title exists

File: components/bounties/[bountyId]/components_v3/BountyHeader.tsx [hunk 7]
```diff
@@ -99,128 +71,54 @@ export default function BountyHeader ({ bounty, permissions, refreshBountyPermis
...
+                  {
+                    permissions?.userPermissions?.lock && isBountyLockable(bounty) && (
+                      [
+                        <Tooltip key='stop-new' arrow placement='right' title={`Prevent new ${bounty.approveSubmitters ? 'applications' : 'submissions'} from being made.`}>
+                          <IconButton
+                            onClick={() => {
+                              closeSubmissionsModal.open();
+                            }}
+                          >
+                            <LockIcon color='secondary' fontSize='small' />
```
Comment (mattcasey) [inline components/bounties/[bountyId]/components_v3/BountyHeader.tsx:87]: can we switch to LockOpen when it is locked?

File: components/[pageId]/DocumentPage/components/BountyProperties.tsx [hunk 8]
```diff
@@ -0,0 +1,447 @@
...
+      </div>
+      <Stack gap={0.5} flexDirection='row' mt={2}>
+        <FormLabel sx={{
+          fontWeight: 500
+        }}
+        >Advanced Settings
+        </FormLabel>
+        <IconButton
+          size='small'
+          onClick={() => {
```
Comment (mattcasey) [inline components/[pageId]/DocumentPage/components/BountyProperties.tsx:244]: can u put thie onClick on the Stack element instead, so clicking Advanced Settings also expands the row?

File: components/bounties/[bountyId]/components/BountyApplyButton.tsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  return !userHasApplied ? (
+    <Tooltip
+      placement='top'
+      title={applyButtonTooltipTitle}
+    >
+      <Box component='span'>
+        <Button
+          disabled={newApplicationsSuspended || !permissions?.userPermissions.work}
+          onClick={onClick}
+        >Apply
```
Comment (mattcasey) [inline components/bounties/[bountyId]/components/BountyApplyButton.tsx:41]: this should just say "Apply to this bounty" and no tooltip unless it's disabled. Also please add 'arrow' to the tooltip

**Comment by mattcasey** [comment]
Ran into a bug, i think. I applied to a bounty, accepted my submission (as the original creator) and now, as applictn, i can only Submit - but i haven't even provided a submission and the input is not there:
image

File: components/bounties/[bountyId]/components/BountyApplyButton.tsx (hunk 9 above)
Comment (Devorein) [reply components/bounties/[bountyId]/components/BountyApplyButton.tsx:41]: Alright done

File: components/[pageId]/DocumentPage/components/BountyProperties.tsx (hunk 8 above)
Comment (Devorein) [reply components/[pageId]/DocumentPage/components/BountyProperties.tsx:244]: Sure thing

File: components/bounties/components/BountyCard.tsx (hunk 6 above)
Comment (Devorein) [reply components/bounties/components/BountyCard.tsx]: Nice catch

File: prisma/schema.prisma (hunk 5 above)
Comment (Devorein) [reply prisma/schema.prisma:398]: After discussion we've decided that its safe to perform this migration on the same PR

File: components/bounties/[bountyId]/components/BountyApplicantList.tsx (hunk 2 above)
Comment (Devorein) [reply components/bounties/[bountyId]/components/BountyApplicantList.tsx:38]: Nice catch. I used that prop at the very beginning but forgot to remove it later

File: components/bounties/[bountyId]/BountyDetails.tsx (hunk 1 above)
Comment (Devorein) [reply components/bounties/[bountyId]/BountyDetails.tsx:33]: You are right. I removed `updateCurrentBountyId`

File: components/bounties/[bountyId]/components_v3/BountyApplicationForm.tsx (hunk 3 above)
Comment (Devorein) [reply components/bounties/[bountyId]/components_v3/BountyApplicationForm.tsx:1]: Not really most of the components were just modifications of existing ones and they were placed inside `_v3` so I didn't think of relocating them

File: components/common/Page/CreatePageButton.tsx (hunk 4 above)
Comment (Devorein) [reply components/common/Page/CreatePageButton.tsx:29]: Good point. I've created separate components for creating a new proposal and a vote

File: components/bounties/[bountyId]/components_v3/BountyHeader.tsx (hunk 7 above)
Comment (Devorein) [reply components/bounties/[bountyId]/components_v3/BountyHeader.tsx:87]: fixed the whole bounty submissions lock flow and updated as per the requirements
